packages/flutter/lib/src/gestures/tap.dart

```diff
@@ -143,6 +145,13 @@ class TapGestureRecognizer extends PrimaryPointerGestureRecognizer {
   void _checkUp() {
     if (_wonArenaForPrimaryPointer && _finalPosition != null) {
       resolve(GestureDisposition.accepted);
+      if (!_wonArenaForPrimaryPointer) {
+        // It is possible that resolve has just recursively called _checkUp (see #12470).
```

Comment:
assert _filePosition is also null here

Reply:
done, though not sure why is this needed here?